espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ActivityHelpers.kt

```diff
@@ -87,6 +91,23 @@ object ActivityHelpers {
             AndroidLogger.logger.info("Staring activity with custom options: $intentOptions")
             makeIntent(intentOptions)
         }
-        instrumentation.startActivitySync(intent)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && activityOptions !=null) {
+            var options = makeActivityOptions(activityOptions)
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                instrumentation.startActivitySync(intent, options.toBundle())
+            } else {
+                instrumentation.targetContext.startActivity(intent, options.toBundle())
+            }
```

Comment:
here can write like (can reduce var/val):

```kotlin
makeActivityOptions(activityOptions).let {
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
        instrumentation.startActivitySync(intent, it.toBundle())
    } else {
        instrumentation.targetContext.startActivity(intent, it.toBundle())
    }
}

```

Reply:
Yes. When i am using this, it will have an error on junit test. Let me try again as you said